tests/h/services/list_groups_test.py

```diff
@@ -113,6 +112,34 @@ def test_sorts_groups_alphabetically(self, svc, user, mixed_groups):
 
         assert [group.pubid for group in results] == expected_sorted_pubids
 
+    def test_returns_associated_groups_if_user(self,
+                                               svc,
+                                               authority,
+                                               user,
+                                               unscoped_restricted_groups,
+                                               scoped_restricted_groups,
+                                               scoped_open_groups,
+                                               unscoped_open_groups,
+                                               private_groups):
+        user.groups = private_groups
+
+        all_groups = scoped_open_groups + unscoped_open_groups + \
+                     scoped_restricted_groups + unscoped_restricted_groups + \
+                     private_groups
+        expected = [group for group in all_groups if group.creator == user or user in group.members]
+
+        results = svc.associated_groups(authority=authority, user=user)
+
+        assert set(results) == set(expected)
+
+    def test_returns_no_associated_groups_if_no_user(self, svc, authority):
```

Comment:
It looks to me as though this test has different setup than the one above in that it doesn't create any groups in the DB before calling `associated_groups`. This means that an implementation of `associated_groups` which always returned every group in the DB would I think have the same result.